presto-main/src/test/java/io/prestosql/sql/query/TestUnnest.java

```diff
@@ -199,4 +199,17 @@ public void testRepeatedExpressions()
                 "SELECT * FROM (VALUES 1) t, UNNEST(ARRAY['a', 'b'], ARRAY['a', 'b']) u (x, y)"))
                 .matches("VALUES (1, 'a', 'a'), (1, 'b', 'b')");
     }
+
+    @Test
+    public void testDereferencesWithUnnest()
+    {
+        assertThat(assertions.query(
```

Comment:
Can you add some more tests?

* One that references all unnested fields
* One that references some fields from both arrays
* One that references a whole row column (e.g., `z` in the query below)